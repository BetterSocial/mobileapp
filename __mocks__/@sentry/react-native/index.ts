export const addBreadcrumb = jest.fn();
