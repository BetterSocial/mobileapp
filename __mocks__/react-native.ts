import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN