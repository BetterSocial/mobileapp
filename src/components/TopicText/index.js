import * as React from 'react';
import {StyleSheet, Text} from 'react-native';

import {fonts} from '../../utils/fonts';
import {COLORS} from '../../utils/theme';

const TopicText = ({
  text,
  navigation = null,
  currentTopic = null,
  goToDetailPage,
  item,
  isShortText
}) => {
  const onClick = () => {
    // Do navigation here
    if (goToDetailPage && navigation) {
      return navigation.push('PostDetailPage', {
        // index: index,
        isalreadypolling: item.isalreadypolling,
        feedId: item.id,
        // refreshParent:  getDataFeedsHandle,
        data: item,
        isCaching: false
      });
    }
    if (!navigation || currentTopic === text.replace('#', '')) return null;
    return navigation.push('TopicPageScreen', {id: `topic_${text.replace('#', '').toLowerCase()}`});
  };

  return (
    <Text
      testID="topicTextComponent"
      onPress={onClick}
      style={[styles.text, styles.shortText(isShortText)]}>
      {text}
    </Text>
  );
};

const styles = StyleSheet.create({
  text: {
    color: COLORS.blue,
    fontFamily: fonts.inter[500]
  },
  shortText: (isShort) => ({
    color: isShort ? 'rgba(255, 255, 255, 0.7)' : COLORS.blue
  })
});

export default React.memo(TopicText);
