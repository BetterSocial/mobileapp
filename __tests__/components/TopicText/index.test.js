import * as React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

import TopicText from '../../../src/components/TopicText'

beforeEach(() => {
    jest.useFakeTimers()
})
describe('Testing Topic Text', () => {
    const navigation = {
        push: jest.fn()
    }
    const text = '#topicPoll'

    it('Match snapshot', () => {
        const tree = renderer.create(<TopicText navigation={navigation} text={text} currentTopic={'poll'} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Will not navigate when navigation is null', () => {
        const { getByTestId } = render(<TopicText text={text} poll={'poll'} />)
        fireEvent.press(getByTestId('topicTextComponent'))
        expect(navigation.push).toBeCalledTimes(0)
    })

    it('Will not navigate when navigation is set and target topic is the same as currentTopic', () => {
        const { getByTestId } = render(<TopicText navigation={navigation} text={text} currentTopic={'topicPoll'} />)
        fireEvent.press(getByTestId('topicTextComponent'))
        expect(navigation.push).toBeCalledTimes(0)
    })

    it('Will navigate when navigation is set and target topic is different as currentTopic', () => {
        const { getByTestId } = render(<TopicText navigation={navigation} text={text} currentTopic={'poll'} />)
        fireEvent.press(getByTestId('topicTextComponent'))
        expect(navigation.push).toBeCalledTimes(1)
    })
})