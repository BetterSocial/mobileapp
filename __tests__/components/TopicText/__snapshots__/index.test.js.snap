// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Topic Text Match snapshot 1`] = `
<Text
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Inter-Medium",
      },
      Object {
        "color": "#2F80ED",
      },
    ]
  }
  testID="topicTextComponent"
>
  #topicPoll
</Text>
`;
